loadtest/control/simulcontroller/actions.go

```diff
@@ -1760,3 +1760,42 @@ func (c *SimulController) updateThreadRead(u user.User) control.UserActionRespon
 
 	return control.UserActionResponse{Info: fmt.Sprintf("updated read state of thread %s", thread.PostId)}
 }
+
+func (c *SimulController) getInsights(u user.User) control.UserActionResponse {
+	team, err := u.Store().CurrentTeam()
+	if errors.Is(err, memstore.ErrTeamStoreEmpty) {
+		return control.UserActionResponse{Info: "no team set"}
+	} else if err != nil {
+		return control.UserActionResponse{Err: control.NewUserError(err)}
+	} else if team == nil {
+		return control.UserActionResponse{Err: control.NewUserError(errors.New("current team should be set"))}
+	}
+
+	userID := u.Store().Id()
+
+	if rand.Float64() < 0.50 {
+		// view user insights
+		if _, err := u.GetTopThreadsForTeamSince(userID, team.Id, "28_day", 0, 100); err != nil {
```

Comment:
Looking at web client's behaviour it seems like we are defaulting to fetch the weekly insights rather than the monthly ones. I'd expect to see the same here with some additional logic to possibly switch between the available options.

Reply:
Please check the new `getInsights` logic?